code/drasil-gool/GOOL/Drasil/ClassInterface.hs

```diff
@@ -594,8 +594,8 @@ convType Double = double
 convType Char = char
 convType String = string
 convType (List t) = listType (convType t)
+convType (Iterator _) = error "Iterators are only available within forEach loops in C++"
```

Comment:
I didn't really know much about convType, so this might be questionable.

Reply:
You should be able to remove the `Iterator` node from the definition of `CodeType`. Then you can remove this whole line from `convType`.

I'm thinking the only place `Iterator` is currently used is in the `cppIterType` function, but you can just remove it there too.

Reply:
I think all traces of `Iterator` should go, so that should make the above obsolete too.

Reply:
Oh I can't do that, cause this function uses CodeType
```haskell
cppIterType :: (RenderSym r) => VSType r -> VSType r
cppIterType = onStateValue (\t -> typeFromData (Iterator (getType t)) 
  (getTypeString t `nmSpcAccess` cppIterator) (stdAccess' (RC.type' t) 
  `nmSpcAccess'` text cppIterator))```